src/local/motion/NonRigidObjectTransformationUsingBSplines.cxx

```diff
@@ -230,11 +230,58 @@ set_deformation_field_from_file(DeformationFieldOnCartesianGrid<3,float>& deform
   return Succeeded::yes;
 }
 
+static
+Succeeded
+set_deformation_field_from_file(DeformationFieldOnCartesianGrid<3,float>& deformation_field,
+                const std::string& deformation_field_multicomponent_filename,
+                CartesianCoordinate3D<float>& grid_spacing,
+                CartesianCoordinate3D<float>& origin)
+{
+    // Read the multicomponent image as a single image to create deformation image of correct size
+    shared_ptr<DiscretisedDensity<3,float> > single_discretised_sptr
+            (read_from_file<DiscretisedDensity<3,float> >(deformation_field_multicomponent_filename));
+    if (is_null_ptr(single_discretised_sptr)) {
+        warning("Error reading %s", deformation_field_multicomponent_filename.c_str());
+        return Succeeded::no;
+    }
+
+    VoxelsOnCartesianGrid<float> const * single_voxels_ptr =
+      dynamic_cast<VoxelsOnCartesianGrid<float> const *>(single_discretised_sptr.get());
+    if (is_null_ptr(single_voxels_ptr)) {
+        warning("Error reading %s: should be of type VoxelsOnCartesianGrid", deformation_field_multicomponent_filename.c_str());
+        return Succeeded::no;
+    }
+
+    deformation_field[1] = *single_voxels_ptr;
+    deformation_field[2] = *single_voxels_ptr;
+    deformation_field[3] = *single_voxels_ptr;
+
+    // Read the multicomponent image
+    VoxelsOnCartesianGrid<CartesianCoordinate3D<float> > voxel_coords =
```

Comment:
creates a unnecessary copy.
can do
```C++
 unique_ptr<> uptr =  read_from_file();
 const VoxelsOnCartesianGrid<CartesianCoordinate3D<float> >& voxel_coords(*uptr);
```
if you want to avoid the `*` all the time